### ISSUE TITLE
☂️ Service Discovery UI

### ISSUE THREAD

**Issue by philrenaud** [hashicorp/nomad#14193]
## A note on Abilities/ACLs:
- Services do not have their own ACL requirements; they depend on `read-job` alone for viewing, and `submit-job` in the case of service de-registration, which the UI will not handle.

## Services Tab (`ui/services`)
NOTE: On hold / de-prioritized for inefficient number of calls that would be required.

## General Service model and fixture updates
- [x] `8` [Model, Serializer and Adapter updates (replace existing fragment; make watchable)](https://github.com/hashicorp/nomad/pull/14226)
- [x] `3` [Mirage fixtures](https://github.com/hashicorp/nomad/pull/14319)
- [x] `5` [E2E tests using Mirage fixtures](https://github.com/hashicorp/nomad/pull/14319)

## Job-contextual Services Tab (`ui/jobs/:JOBID/services` and `ui/jobs/:JOBID/services/:SERVICEID`)
- [x] `3` [Scaffold a jobs/job/services route](https://github.com/hashicorp/nomad/issues/14202)
  - [x] `1` [Aggregate allocations by Alloc ID](https://github.com/hashicorp/nomad/issues/14213)
- [x] `1` [Scaffold new child route](https://github.com/hashicorp/nomad/pull/14226)
- [x] `0` [Add new LinkTo to Job subnav](https://github.com/hashicorp/nomad/pull/14226)
- [x] `1` ~If job spec has any consul services, warn them on the jobs/services page that we don't show those there, and link to docs.~ nvm! [Can show Consul services, too!](https://github.com/hashicorp/nomad/pull/14226)

## Allocation-contextual Services Table (`ui/allocations/:ALLOCID`)
- [x] `3` Health Checks (`v1/allocation/:ALLOCID/checks`)
- [x] `1` Aggregate check status for a given service into a health check bar
- [x] STRETCH `3` Merge existing services that come back with the allocation, and may include Consul services, with Nomad ones that come back from /checks)
- [x] [handle visualizing health status checks for duplicate services](https://github.com/hashicorp/nomad/issues/14380)
- [x] `2` [Show Task-level services on the alloc page](https://github.com/hashicorp/nomad/issues/14198)

## Allocation-contextual Service-specific Flyout (`ui/allocations/:ALLOCID?service=:SERVICEID`)
- [x] `1` [Table setup](https://github.com/hashicorp/nomad/issues/14386)
- [x] `2` [Format error output for failing states](https://github.com/hashicorp/nomad/issues/14386)
- [x] `1` [Query Parameters for reload etc.](https://github.com/hashicorp/nomad/issues/14386)
- [ ] #14482

**Comment by philrenaud** [comment]
> You mean acceptance tests? Or E2E in Playwright?

Acceptance tests

**Comment by ChaiWithJai** [comment]
> Health Checks (v1/allocation/:ALLOCID/checks)

It appears that this block of work must occur before "Aggregate check status for a given service into a health check bar" can happen and before any work in "Allocation-contextual Service-specific Flyout" can happen.

**Comment by ChaiWithJai** [comment]
> STRETCH 3 Merge existing services that come back with the allocation, and may include Consul services, with Nomad ones that come back from /checks)

Should we mark this as not possible?

**Comment by philrenaud** [comment]
> > STRETCH 3 Merge existing services that come back with the allocation, and may include Consul services, with Nomad ones that come back from /checks)

> Should we mark this as not possible?

I think this is actually the default way that things will work; we use the current list to show both consul and nomad services, and we hydrate the nomad ones with health check statuses.
